Update .user.ini when setting upload size limit

RobinMcCorkell opened the pull request:
We try to change as many files as possible, so if we can write to `.user.ini` but not to `.htaccess` then we will (but the file contents will become inconsistent). Of course, whichever one actually affects the installation will be shown in the upload limit box in the admin settings, so there is no ambiguity in what happened, and the logs clearly state if any files could not be written to.

Not sure if this can be unit tested?

Fixes #14748

cc @PVince81 @LukasReschke @DeepDiver1975 @RealRancor @MorrisJobke

Commit:
Update .user.ini when setting upload size limit

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/d9d40e68-9b4e-494e-9ff8-548a6f99d7e8) was created.

mmattel commented:
:+1:

karlitschek commented:
@DeepDiver1975 What do you think?

MorrisJobke commented on lib/private/files.php:
```diff
@@ -274,38 +275,63 @@ static function setUploadLimit($size) {
-				$htaccess .= "\n" . $setting;
+
+			$content = '';
+			while (!feof($handle)) {
```
Mayber use file_get_contents: https://secure.php.net/manual/de/function.file-get-contents.php

MorrisJobke commented on lib/private/files.php:
```diff
@@ -274,38 +275,63 @@ static function setUploadLimit($size) {
+			rewind($handle);
+			fwrite($handle, $content);
+
+			fclose($handle);
```
Use file_put_contents instead of the 4 lines above: https://secure.php.net/manual/de/function.file-put-contents.php

DeepDiver1975 commented:
> Not sure if this can be unit tested?

I guess it is possible
- by passing in the root folder where files are to be read/written this can be tweaked for unit testing.

Xenopathic commented:
@MorrisJobke I used `fopen()` and related functions so we only need to open the file and check for access once - before we do any reading or replacing of the contents. The original code used `file_get_contents()` and `file_put_contents()`.

Commit:
Unit test OC_Files::setUploadLimit()

There was also a bug with checking the upper limit on the passed upload
size. PHP does funny things with integer vs float comparisons, so our
check didn't work. Now the check is much simpler, and ensures the value
is sane.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/owncloud/core/inspections/0a591e4c-c051-40e7-9fca-d5485ea70245) was created.

Xenopathic commented:
Unit tests! There was also a bug with checking the upper limit on the passed upload
size. PHP does funny things with integer vs float comparisons, so our
check didn't work. Now the check is much simpler, and ensures the value
is sane.

Xenopathic commented:
@DeepDiver1975 @MorrisJobke Re-review please?

Xenopathic commented:
@MorrisJobke Do you still think this should be using `file_get_contents()` and `file_put_contents()` instead of `fopen()`?

MorrisJobke commented:
> @MorrisJobke Do you still think this should be using file_get_contents() and file_put_contents() instead of fopen()?

I'm fine with the way it is now.

Xenopathic commented:
@DeepDiver1975 @MorrisJobke please review

MorrisJobke commented:
Tested and works :+1: (also with changed permissions)
